uxarray/_scrip.py

```diff
@@ -1,11 +1,142 @@
+    in_ds : :class:`xarray.Dataset`
+        Original scrip dataset of interest being used
+
+    outfile : :class:`xarray.Variable`
+        file to be returned by _populate_scrip_data, used as an empty placeholder file
+        to store reassigned SCRIP variables in UGRID conventions
     """
-    warn("Function not implemented yet. FYI, attempted to read SCRIP file: " +
-         str(filepath))
-    # TODO: create ds
+
+    if in_ds['grid_area'].all():
+
+        # Create Mesh2_node_x/y variables from grid_corner_lat/lon
+        # Turn latitude scrip array into 1D instead of 2D
+        corner_lat = in_ds['grid_corner_lat'].stack(y=("grid_size",
+                                                       "grid_corners"))
+
+        # Return only the index of unique values to preserve order
+        lat_idx = np.unique(corner_lat.values, return_index=True)[1]
+        sort_lat = sorted(lat_idx)
+
+        # Create placeholder array for the sorted unique values
+        unique_lat = np.zeros(len(sort_lat))
+        # unique_lat = corner_lat[sort_lat]
+        for i in range(len(sort_lat) - 1):
+            unique_lat[i] = (corner_lat[sort_lat[i]])
+
+        # Repeat above steps with longitude data instead
+        corner_lon = in_ds['grid_corner_lon'].stack(x=("grid_size",
+                                                       "grid_corners"))
+        lon_idx = np.unique(corner_lon, return_index=True)[1]
+        sort_lon = sorted(lon_idx)
+
+        unique_lon = np.zeros(len(sort_lon))
+        for i in range(len(sort_lon) - 1):
+            unique_lon[i] = (corner_lon[sort_lon[i]])
+
+        # Create Mesh2_node_x/y from unsorted, unique grid_corner_lat/lon
+        outfile['Mesh2_node_x'] = unique_lon
+        outfile['Mesh2_node_y'] = unique_lat
+
+        # Create Mesh2_face_x/y from grid_center_lat/lon
+        outfile['Mesh2_face_x'] = in_ds['grid_center_lon']
+        outfile['Mesh2_face_y'] = in_ds['grid_center_lat']
+
+        # Create array using matching grid corners to create face nodes
+        # find max face nodes
+        max_face_nodes = 0
+        for dim in in_ds.dims:
+            if "grid_corners" in dim:
+                if in_ds.dims[dim] > max_face_nodes:
+                    max_face_nodes = in_ds.dims[dim]
+
+        # create an empty conn array for storing all blk face_nodes_data
+        conn = np.empty((len(unique_lon), max_face_nodes))
+
+        # set the face nodes data compiled in "connect" section
+        outfile["Mesh2_face_nodes"] = xr.DataArray(
+            data=(conn[:] - 1),
+            dims=["nMesh2_face", "nMaxMesh2_face_nodes"],
+            attrs={
+                "cf_role":
+                    "face_node_connectivity",
+                "_FillValue":
+                    -1,
+                "start_index":
+                    np.int32(
+                        0
+                    )  # NOTE: This might cause an error if numbering has holes
+            })
+        # face_arr = []
```

Comment:
@erogluorhan I left the original code here mainly for discussion and references purposes. I'm having a little bit of difficulty understanding how to determine the values for the `nMesh2_face_nodes` part of the `Mesh2_face_nodes` variable. From the UGRID website ([for 2D flexible mesh](https://ugrid-conventions.github.io/ugrid-conventions/#:~:text=the%20current%20convention.-,Example%3A,-dimensions%3A%0AnMesh2_node%20%3D%205)) there isn't much to go off of. When looking at `outCSne30.ug`, I also can't seem to place how the data points being selected and placed for this array. Just a quick print out of each so you can see what I am seeing:
Mesh2_node_x:
`array([315.      ,  45.      , 135.      , ..., 141.09969 , 138.033171,
       135.      ])`

Mesh2_node_y:
`array([-35.26439 , -35.26439 , -35.26439 , ...,  40.837604,  39.549078,
        38.143314])`

Mesh2_face_nodes:
`array([[   0.,    8.,  356.,  124.],
       [   8.,    9.,  357.,  356.],
       [   9.,   10.,  358.,  357.],
       ...,
       [5399., 5400.,  299.,  300.],
       [5400., 5401.,  298.,  299.],
       [5401.,  297.,    6.,  298.]])`

It isn't the first two indexes of each row that are giving me much pause, but rather the last two (which should be coming from Mesh2_node_x and Mesh2_node_y) that are, as they don't follow a strict order that would match with their inputs like the first two do.

Reply:
I am assuming this is a comment to my comment above that starts with "The dimensionality of this variable does not ...". 

The UGRID website ([for 2D flexible mesh](https://ugrid-conventions.github.io/ugrid-conventions/#:~:text=the%20current%20convention.-,Example%3A,-dimensions%3A%0AnMesh2_node%20%3D%205)) says that "The attribute `face_node_connectivity` points to an index variable identifying for every face the indices of its corner nodes." So, in our case, we are naming that index variable as "Mesh2_face_nodes", which is also the example name given in the same website (Though, if it was a UGRID file reading task, we would first. look at the attribute `face_node_connectivity` and it would give us the exact name of the variable that is stored in that netCDF file). 

From the statement, we understand that `Mesh2_face_nodes` should identify for every face the indices of its corner nodes, i.e. it should have a size of `nMesh2_face` * `nMaxMesh2_face_nodes`. That's to say, we should have as many rows as the number of faces, and in each row, we should have the index of nodes that are on that specific face. More clearly from your example:

Mesh2_face_nodes:
`array([[ 0., 8., 356., 124.], [ 8., 9., 357., 356.], [ 9., 10., 358., 357.], ..., [5399., 5400., 299., 300.], [5400., 5401., 298., 299.], [5401., 297., 6., 298.]])`

That means the face #0 has the nodes with index numbers 0, 8, 356, 124 as its corners; the face #1 has the nodes with index numbers 8, 9, 357, 356 in it, etc. 

A small side-note here, even though your provided `Mesh2_face_nodes` array values look like float (e.g. "356.", "124.", all numbers with a period at the end), if you check its data type, it is int32 instead, because it only holds node indices. 

In your SCRIP case, SCRIP files do not keep track of indices for nodes and faces. So, you should be generating and keeping tracking of indices  for your faces and nodes when you are creating a UGRID data structures from a SCRIP file.

I hope this helps.